apps/andi/lib/andi/services/dataset_retrieval_service.ex

```diff
@@ -14,4 +16,24 @@ defmodule Andi.Services.DatasetRetrieval do
       {:error, reason} -> raise reason
     end
   end
+
+  def update(%SmartCity.Dataset{} = dataset) do
+    Brook.ViewState.merge(@collection, dataset.id, dataset)
+  end
+
+  def get!(id, collection \\ @collection) do
+    Brook.get!(instance_name(), collection, id)
+  end
+
+  def get_all_dataset!(instance \\ instance_name()) do
+    Brook.get_all_values!(instance_name(), @collection)
+  end
+
+  def get_all_ingested_time!() do
+    Brook.get_all_values!(instance_name(), :ingested_time)
+  end
+
+  def delete(id) do
+    Brook.ViewState.delete(@collection, id)
+  end
```

Comment:
Why do these functions on Brook.ViewState NOT take in an instance of Brook?  I'd like to understand how this works to make sure we're calling these correctly.

Reply:
Also, can we make all of these function signatures consistent?  Either have ALL of them take in the optional instance of Brook, or have NONE of them take in this arg.